perllib/FixMyStreet/DB/ResultSet/Nearby.pm

```diff
@@ -23,8 +23,17 @@ sub nearby {
         if $args{ids};
     $params->{'problem.category'} = $args{categories} if $args{categories} && @{$args{categories}};
 
-    $params->{$c->stash->{report_age_field}} = { '>=', \"current_timestamp-'$args{report_age}'::interval" }
-        if $args{report_age};
+    my $report_age = $args{report_age};
+    if ( $report_age && ref $report_age eq 'HASH' ) {
+        $params->{'-or'} = FixMyStreet::DB::ResultSet::Problem->report_age_subquery(
```

Comment:
There's an issue here because `non_public_if_possible` also sets `-or` directly. We'll need to have both work in a way that they can work with the other.